packages/clerk-js/src/ui/hooks/useNavigate.ts

```diff
@@ -3,6 +3,7 @@ import React from 'react';
 import { useCoreClerk } from '../contexts';
 import { RouteContext } from '../router';
 
+// FIXME: When you remove this return type and use TypeScript inference, it will show errors throughout the codebase.
```

Comment:
You might already know this, but thought it worth nothing 🤷🏻‍♂️.

When you remove the defined return type and use TypeScript inference it appears that the `useNavigate` usage in codebase needs some refactoring. It might just be a type issue, but in one case some urls props/context can be undefined [here](https://github.com/clerkinc/javascript/blob/main/packages/clerk-js/src/ui/contexts/ClerkUIComponentsContext.tsx#L192-L196). The build output with the removed return type for `useNavigate had a few TypeScript errors.

<img width="1662" alt="CleanShot 2023-04-20 at 19 01 39@2x" src="https://user-images.githubusercontent.com/4893048/233460779-52a52fe9-967a-4342-86ec-1e8608e552f8.png">